### PR TITLE
Reorganizing and inflating the entire Hackinator.

### PR DESCRIPTION
**Pull request by Enlight1**
I renamed the old 'weapons' folder to a new and possibly more-fitting 'bullets' folder, and then made a customizer for each of the 10 guns.

Plus added some lightning to a renamed "spike_damage.xml" and made a XML for other bullet-related NPC's and such.

You can look at the commits in my fork if you want to see the step-by-step changes, since the moved files are listed as line deletions and additions.

One issue is I did not explain Bullet Families anywhere (Pixels weird way of having the maximum-bullet limit work for pairs of 3). Could take suggestions for noting that somewhere and any other issues.

### CODE REVIEW

**Comment by Enlight1** [comment]
So I realized I forgot to add the "modes" or whatever the modifiers are called for the effects, which are all literally 2 bytes before the effects being pushed. Oops.

I've made an effect list just like my bullet list so I guess I'll write up a help file for those, and explain the bullet families in there as well. We could maybe hold off until that's done.

But other than that, @taedixon what's your thoughts?

**Comment by taedixon** [comment]
theres a lot of changes here, and it's pretty good overall. I definitely agree that "bullet family" needs some kind of explanation (As far as I can tell it's set up  to group bullets by their respective weapons but it's not obvious at all that changing the IDs fired by the weapon changes the bullet families you need to check; as well if you use bullets from multiple "families" it will not be possible to manage fire rate this way).

as well some of the language on the fields doesn't immediately betray their purpose, for example "Effect spawning" may not immediately be obvious that it refers to the "muzzle flash / related" effect that's generated when you fire a weapon.

And, is the blade really supposed to be so skimpy compared to the rest? :P

File: hacks/weapons/missiles.xml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+				<field type="label">Second Family:</field>
+				<field type="label">Second Family Shot Limit*:</field>
+				<field type="text" size="1" offset="0x41E92C" col="1"></field>
+				<field type="text" size="1" offset="0x41E937"></field>
+				<field type="text" size="1" offset="0x41E940"></field>
+				<field type="text" size="1" offset="0x41E94B"></field>
+			</panel>
+			<panel>
+				<field type="label">*Plus 1, so "1" actually means "2"</field>
+				<field type="label">†Remember: Each shot creates 3 bullets</field>
```
Comment (taedixon) [inline hacks/weapons/missiles.xml]: This dagger doesn't seem to relate to any particular field..?

File: hacks/weapons/supermissiles.xml [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+				<field type="label">Second Family:</field>
+				<field type="label">Second Family Shot Limit*:</field>
+				<field type="text" size="1" offset="0x41E860" col="1"></field>
+				<field type="text" size="1" offset="0x41E86B"></field>
+				<field type="text" size="1" offset="0x41E874"></field>
+				<field type="text" size="1" offset="0x41E87F"></field>
+			</panel>
+			<panel>
+				<field type="label">*Plus 1, so "1" actually means "2"</field>
+				<field type="label">†Remember: Each shot creates 3 bullets</field>
```
Comment (taedixon) [inline hacks/weapons/supermissiles.xml]: here as well

**Comment by Enlight1** [comment]
Yeah, will definitely make some help files for the bullet and effect lists, and in there I can explain the families (as well as the modifiers for the effects when I add them). Unless it gets entirely recoded into something more edit-friendly, will just have to carefully explain the limitation.

I'll see about rewording things to be more clear. That is one of the bigger issues because I basically threw everything I could find into one XML per weapon, whereas stuff like the 12 effect values mainly take up space when I could imagine most people need not touch the stuff, but likewise I don't want to inflate the hackinator with tons of XML's either.

And the blade is unfortunately rather basic. I'll do a quick rehaul of the visual layout and double-check if I missed anything, but it doesn't have effects or sounds or anything fun.

Oh, and those daggers were for lvl 3 maximum shots, I must have accidently deleted them when I relabeled.

So yeah give me a couple days and I'll see what I can do.

**Comment by Enlight1** [comment]
I'm back @taedixon . I'm basically gonna redo this pull request since we've both been working independently and I have 40ish new XML's gathered from the entire CSMC discord server working together. The entire hacks folder is about to be updated.

**Comment by Enlight1** [comment]
And that's it for today. Gonna try to be more regular and open in the future instead of hoarding these up. No promises though. There are some unfinished XML's in there (fully functional but missing something) and I might revamp the help section sometime too. Let's work together before the folder gets any bigger.

**Comment by Enlight1** [comment]
So @taedixon just this morning BLink noticed that Door's titlescreen hack has had a bug for 3 years now, treating the Char X Render Pos. as a word instead of a byte (causing crashes if you edited it). Surprised nobody noticed all these years, right under all of our noses.

File: hacks/TSC modifications/tsc+_timg.xml [hunk 3]
```diff
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<hack name="TSC+ TIMG addon">
+<hack name="TSC+ TIMG addon" author="Noxid">
 	<panel>
 		<field type="info">
 		 An addon for TSC+ that lets you use TIMG commands.
-		 Note that this also modifies the title screen behaviour.
+		 Note that this also modifies the title screen behavior.
```
Comment (taedixon) [inline hacks/TSC modifications/tsc+_timg.xml]: fight me, american pig-dog

File: hacks/misc editors/text_colour.xml [hunk 4]
```diff
@@ -1,9 +1,10 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<hack name="Text Colour">
+<hack name="Text Colour" author="Noxid?">
```
Comment (taedixon) [inline hacks/misc editors/text_colour.xml]: yeah I think this is something I added after HACKER STORY

File: hacks/player properties/player_rendersize_recoded.xml [hunk 5]
```diff
@@ -0,0 +1,256 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<hack name="Player Size" author="SlightlyInteligentMonkey">
```
Comment (taedixon) [inline hacks/player properties/player_rendersize_recoded.xml]: ive seen this spelled with 1 `l` or 2 in different files so I don't really know which is right but might wanna check em over

File: hacks/weapons/custom/shotgun.xml [hunk 6]
```diff
@@ -0,0 +1,89 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<hack name="Shotgun Hack" author="Noxid">
+	<panel>
+		<field type="label">REPLACES MACHINE GUN</field>
+		<field type="info">
+		This turns the Machine Gun and the MG bullets to become a shotgun. Knockback
+		is applied to the player when fired. Refer to /hacks/bitmap/shotgun_example.bmp
```
Comment (taedixon) [inline hacks/weapons/custom/shotgun.xml]: i should add an "image" element to the xml structure... but not today :3

File: hacks/NPCs/recoded vanilla/balfrog_editing.xml [hunk 7]
```diff
@@ -0,0 +1,62 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<hack name="Balfrog stuff" author="SlightlyInteligentMonkey">
```
[the remark this answers is not in the document]
Comment (Enlight1) [reply hacks/NPCs/recoded vanilla/balfrog_editing.xml]: Yep, there was probably a point we all put typos in our nicknames on discord and I accidently copy/pasted them. It's two L's on all files.

File: hacks/TSC modifications/tsc+_timg.xml (hunk 3 above)
Comment (Enlight1) [reply hacks/TSC modifications/tsc+_timg.xml]: Blame notepad+

File: hacks/weapons/custom/shotgun.xml (hunk 6 above)
Comment (Enlight1) [reply hacks/weapons/custom/shotgun.xml]: that would help with overcoming the weird /bitmap/ folder thing.

**Comment by Enlight1** [comment]
> why do a bunch / all of the files have the +x (executable) permission set now? that shouldn't be there.
Gonna be honest, I 100% have no idea what you are talking about, was probably some messup from the files transferring across my two computers and figuring out github. Could you explain what this is and how to fix it?

But yeah, I fixed all the issues you stated. I might actually finish `\misc editors\weapon_effects.xml` or `\bullets\vanilla editing\snake.xml` . But it's more likely I'll procrastinate and forget.

Also if you want to put Door's recent XML somewhere if you end up pulling, or if there is a way I can update my repository to do that without issues, I wouldn't know.

**Comment by taedixon** [comment]
if you notice many files have `100644 → 100755` on them in the change list. These indicate file permissions, with `644` meaning "owner can read/write, anyone can read". `755` means "owner can read, write, or execute this file, anyone can read or execute this file". It's not like, going to break anything but it's a weird thing to have changed and i'd prefer if it was fixed. pretty sure it's not a windows thing because most of the initial work on BL was done under windows.

**Comment by Enlight1** [comment]
I installed git and used a lot of googling to figure it out.

**Pull request merged**